### ISSUE TITLE
No error handling present in GOV.UK Frontend review application

### ISSUE THREAD

**Issue by nickcolley** [alphagov/govuk-frontend#1397]
This issue is from a May 2019 external accessibility audit report.

**Issue ID**: DAC_Issue13
**URL**: http://govuk-frontend-v3.herokuapp.com/full-page-examples/update-your-account-details
**Page**: Update your account details
**WCAG Reference**: [3.3.1 Error Identification (Level A)](https://www.w3.org/WAI/WCAG21/Understanding/error-identification.html)

## Screen Shot
![](https://user-images.githubusercontent.com/2445413/58706833-fe8f6480-83aa-11e9-9fd9-831aa50ed950.png)

No error handling was present on the page to inform users that they had entered information incorrectly.
A format was also not provided for the password field. It may be the case that any password format can be used, however, it is important to ensure that if a format IS required, this is stated within hint text above the form field.

## Current Code Ref(s)
```html
<input class="govuk-input" id="email" name="email" type="email" autocomplete="email" spellcheck="false">
```

## Screen reader comments
> "I entered an incorrect format into the e-mail field and was not given information relating to which format the password should be entered into the relevant field. I selected the ‘Save’ button and was taken to a success page. No error handling appeared to alert me that either the password or e-mail address was in an incorrect format. It would benefit users if hint text could be related to the password field and if error handling could appear relating to an incorrect format. This issue was present using JAWS, NVDA and VoiceOver for iPhone and Mac.”

## Solution
Ensure that error handling is present to identify issues with user input. This must include the format required for an email address and where a password format is required also.
An example can be found at: https://design-system.service.gov.uk/patterns/email-addresses/#error-messages

**Comment by nickcolley** [comment]
I've assigned priority low since this behaviour is in our review application which is intentionally not a real world application.

My suggestion would be to add a basic check for the `@` symbol, which would not take too long.

**Comment by nickcolley** [comment]
Fixed in https://github.com/alphagov/govuk-frontend/pull/1418

**Comment by nickcolley** [comment]
Confirmed as fixed in a re-test by DAC on 7th October 2019

**Comment by nickcolley** [comment]
Confirmed as fixed in a re-test by DAC on 7th October 2019
